### ISSUE TITLE
update dependencies in conda-forge feedstock

### ISSUE THREAD

**Issue by CagtayFabry** [asdf-format/asdf#1096]
With the recent releases the conda-forge recipe over at https://github.com/conda-forge/asdf-feedstock has become slightly out of sync with regard to the package dependencies leading to some possible broken installs (see https://github.com/asdf-format/asdf-standard/issues/301)

The requirements in [meta.yaml](https://github.com/conda-forge/asdf-feedstock/blob/master/recipe/meta.yaml#L19-L33)
```yaml
requirements:
  host:
    - pip
    - python >=3.6
    - setuptools
    - setuptools_scm
  run:
    - python >=3.6
    - pyyaml >=3.10
    - jmespath >=0.6.2
    - Jsonschema >=3.0.2,<4
    - numpy >=1.10
    - semantic_version >=2.8
    - importlib_resources >=3
    - packaging >=16.0
```
are not up to date with the release metadata
https://github.com/asdf-format/asdf/blob/3483bfe273205eac39ca20097b2af7a27a3283be/setup.cfg#L24-L37

when working with conda recipes we have made good experience with including `pip check` as a simple test step in `meta.yaml` to make sure the install covers the requirements of the python package, see [here](https://github.com/conda-forge/weldx-feedstock/blob/612f94c315a11bb1e4c7f6122ba0a693440e6864/recipe/meta.yaml#L41-L50) for an example

**Comment by WilliamJamieson** [comment]
This will be fixed when we fix issue #1093. It will require releases of several of our packages and setting up of conda-forge feedstocks in order to do so.

**Comment by CagtayFabry** [comment]
ok, thank you for the update
it is a bit unfortunate that 2.10 was released on conda-forge without updating the dependencies since any normal install of asdf (without excluding 2.10) is now broken

It is also possible to "fix" the current 2.10 release on conda-forge by updating the recipe with a new build number (this is not a new release)
However I am not sure the CI will pass without the `asdf-standard` package also available on conda first

**Comment by WilliamJamieson** [comment]
> ok, thank you for the update it is a bit unfortunate that 2.10 was released on conda-forge without updating the dependencies since any normal install of asdf via conda (without excluding 2.10) is now broken
> 
> It is also possible to "fix" the current 2.10 release on conda-forge by updating the recipe with a new build number (this is not a new release) However I am not sure the CI will pass without the `asdf-standard` package also available on conda first

I honestly don't know much about conda-forge, as I have not used it before starting work on  asdf (nor do I typically use conda). It is my fault that the recipe failed to get updated (I apologize for my naivete leading to this problem). I will see if I can figure out how to fix asdf 2.10 on conda-forge and report back.

**Comment by CagtayFabry** [comment]
No worries, the conda recipes can be quite complex (but are easier to fix than pypi thankfully)

I have opened a draft PR with the changes I think would be required, updating the dependencies to match `setup.cfg` https://github.com/conda-forge/asdf-feedstock/pull/50

**Comment by WilliamJamieson** [comment]
I have fixed conda-forge/staged-recipes#18141 using the new asdf-standard [1.0.1 release](https://github.com/asdf-format/asdf-standard/releases/tag/1.0.1). I have also created the asdf-transform-schemas [0.2.1 release](https://github.com/asdf-format/asdf-transform-schemas/releases/tag/0.2.1), which is being used for the conda-forge/staged-recipes#18146, which will hopefully pass once the asdf-standard feedstock is accepted. I think once both of these recipes have been accepted we can merge your conda-forge/asdf-feedstock#50 PR.

**Comment by WilliamJamieson** [comment]
@CagtayFabry I think the conda-forge issues have been resolved with the merger of conda-forge/asdf-feedstock#50 and conda-forge/asdf-feedstock#51. Please reopen this issue and tag me if you are still having problems.
